### PR TITLE
Adding support for more redis cluster params 

### PR DESCRIPTION
**Pull request by dom-nie**
#### Pull Request (PR) description
Introducing following parameters for redis cluster:
-cluster-slave-validity-factor 
-cluster-migration-barrier
-cluster-require-full-coverage

#### This Pull Request (PR) fixes the following issues
None

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -241,6 +250,9 @@
   Boolean $cluster_enabled       = $::redis::params::cluster_enabled,
   $cluster_config_file           = $::redis::params::cluster_config_file,
   $cluster_node_timeout          = $::redis::params::cluster_node_timeout,
+  $cluster_slave_validity_factor = $::redis::params::cluster_slave_validity_factor,
```
Comment (bastelfreak) [inline manifests/init.pp]: can you please add datatypes to all new parameters and align the `=`?

File: manifests/instance.pp [hunk 2]
```diff
@@ -203,6 +212,9 @@
   Boolean $cluster_enabled       = $::redis::cluster_enabled,
   $cluster_config_file           = $::redis::cluster_config_file,
   $cluster_node_timeout          = $::redis::cluster_node_timeout,
+  $cluster_slave_validity_factor = $::redis::cluster_slave_validity_factor,
```
Comment (bastelfreak) [inline manifests/instance.pp]: same as above

File: manifests/init.pp (hunk 1 above)
Comment (dom-nie) [reply manifests/init.pp]: Done

File: manifests/instance.pp (hunk 2 above)
Comment (dom-nie) [reply manifests/instance.pp]: Done

File: manifests/init.pp [hunk 3]
```diff
@@ -241,6 +250,9 @@
   Boolean $cluster_enabled       = $::redis::params::cluster_enabled,
   $cluster_config_file           = $::redis::params::cluster_config_file,
   $cluster_node_timeout          = $::redis::params::cluster_node_timeout,
+  Integer $cluster_slave_validity_factor = $::redis::params::cluster_slave_validity_factor,
```
Comment (bastelfreak) [inline manifests/init.pp]: Please align the `=` for all parameters. Can you also define the lowest/highlest allowed value for the integer parameters? At least the lowest value should be enforced. This is probably 0 or 1?

Comment (dom-nie) [reply manifests/init.pp]: Done.
Both of them have minimum from 0, I have set cluster_migration_barrier default to 1 as per redis docs

**Pull request merged**